### PR TITLE
Issue #2783: Add snackbar to collection deletion

### PR DESCRIPTION
**Pull request by rocketsroger**
### Pull Request checklist

- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by rocketsroger** [comment]
https://user-images.githubusercontent.com/6579541/127227119-335f33bf-3931-4c18-a59c-b1e76a71abcd.mp4

File: app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt [hunk 1]
```diff
@@ -1011,10 +1011,10 @@ class SmokeTest {
         }.expandCollection(collectionName) {
             clickCollectionThreeDotButton()
             selectDeleteCollection()
-            confirmDeleteCollection()
```
Comment (eliserichards) [inline app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt:1014]: Should this move below as well? Or is `verifySnackbarText` doing that for us?

File: app/src/main/java/org/mozilla/fenix/components/TabCollectionStorage.kt [hunk 2]
```diff
@@ -68,6 +69,15 @@ class TabCollectionStorage(
         }
     }
 
+    suspend fun createCollection(tabCollection: TabCollection): Long? {
```
Comment (eliserichards) [inline app/src/main/java/org/mozilla/fenix/components/TabCollectionStorage.kt:72]: This is clever 👍

File: app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt [hunk 3]
```diff
@@ -706,34 +705,25 @@ class HomeFragment : Fragment() {
...
-                    dialog.dismiss()
-                }
-            }
-            create()
-        }.show()
+    @VisibleForTesting
+    internal fun removeCollectionWithUndo(tabCollection: TabCollection) {
+        val snackbarMessage = getString(R.string.snackbar_collection_deleted)
+
+        lifecycleScope.allowUndo(
```
Comment (eliserichards) [inline app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt:712]: I've never used `lifecycleScope.allowUndo` before. Will this cause any issues with blocking the main thread? should it be on IO instead?

File: app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt [hunk 4]
```diff
@@ -1145,5 +1135,8 @@ class HomeFragment : Fragment() {
         private const val FADE_ANIM_DURATION = 150L
         private const val CFR_WIDTH_DIVIDER = 1.7
         private const val CFR_Y_OFFSET = -20
+
+        // Elevation for undo toasts
+        internal const val ELEVATION = 80f
```
Comment (eliserichards) [inline app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt:1140]: Nit: 

```suggestion
        
        internal const val TOAST_ELEVATION = 80f
```

File: app/src/main/res/values-gu-rIN/strings.xml [hunk 5]
```diff
@@ -760,8 +760,7 @@
     <string name="delete_tab_and_collection_dialog_title">%1$s ને કાઢી નાખીએ?</string>
     <!-- Tab collection deletion prompt dialog option to delete the collection -->
     <string name="tab_collection_dialog_positive">કાઢી નાખો</string>
-    <!-- Tab collection deletion prompt dialog option to cancel deleting the collection -->
-    <string name="tab_collection_dialog_negative">રદ કરો</string>
+    <!-- Tab collection deletion promp dialog option to cancel deleting the collection -->
```
Comment (eliserichards) [inline app/src/main/res/values-gu-rIN/strings.xml:763]: Missed this one!

File: app/src/main/res/values-ta/strings.xml [hunk 6]
```diff
@@ -938,8 +938,7 @@
     <string name="delete_tab_and_collection_dialog_title">%1$s அழிக்கவா?</string>
     <!-- Tab collection deletion prompt dialog option to delete the collection -->
     <string name="tab_collection_dialog_positive">அழி</string>
-    <!-- Tab collection deletion prompt dialog option to cancel deleting the collection -->
-    <string name="tab_collection_dialog_negative">ரத்து</string>
+    <!-- Tab collection deletion prompt dialog option to canceldeleting the collection -->
```
Comment (eliserichards) [inline app/src/main/res/values-ta/strings.xml:941]: And here

**Review by eliserichards — CHANGES_REQUESTED** [review]
I can't remember if the snackbars are usually shown overtop of the toolbar or above it?

File: app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt (hunk 1 above)
Comment (rocketsroger) [reply app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt:1014]: Yes, `confirmDeleteCollection()` works with delete dialogue.  `verifySnakbarTest` works with the newly added snackbar.

File: app/src/main/res/values-gu-rIN/strings.xml (hunk 5 above)
Comment (rocketsroger) [reply app/src/main/res/values-gu-rIN/strings.xml:763]: Good catch.  Thanks

File: app/src/main/res/values-ta/strings.xml (hunk 6 above)
Comment (rocketsroger) [reply app/src/main/res/values-ta/strings.xml:941]: Thanks!

File: app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt (hunk 4 above)
Comment (rocketsroger) [reply app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt:1140]: Agreed.  Thanks

File: app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt (hunk 3 above)
Comment (rocketsroger) [reply app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt:712]: No, this won't block the main thread.  The only part that could block the main thread is end of `removeCollectionWithUndo` and it's launched in IO thread.  Thanks,

**Review by eliserichards — APPROVED** [review]
Looks good! 🚢

**Pull request merged**